java/arcs/core/entity/VariableEntityBase.kt

```diff
@@ -0,0 +1,50 @@
+package arcs.core.entity
+
+import arcs.core.common.Referencable
+import arcs.core.data.FieldName
+import arcs.core.data.RawEntity
+import arcs.core.data.Schema
+
+/**
+ * A base [Entity] to access data from type variables.
+ *
+ * This class behaves just like [EntityBase], except for (de)serialization. During deserialization,
+ * all the fields from [RawEntity] are stored for later use in serialization.
+ *
+ * In this way, an entity representing a type variable will pass data through the system without
+ * a specific description of the data (an exact match with a [Schema]).
+ */
+open class VariableEntityBase(
+    entityClassName: String,
+    schema: Schema,
+    entityId: String,
+    creationTimestamp: Long,
+    expirationTimestamp: Long
+) : EntityBase(entityClassName, schema, entityId, creationTimestamp, expirationTimestamp) {
+
+    private val rawSingletons = mutableMapOf<FieldName, Referencable?>()
+    private val rawCollections = mutableMapOf<FieldName, Set<Referencable>>()
+
+    override fun serialize(): RawEntity {
+        val rawEntity = super.serialize()
+        return rawEntity.copy(
+            singletons = rawSingletons + rawEntity.singletons,
+            collections = rawCollections + rawEntity.collections
+        )
+    }
+
+    override fun deserialize(
```

Comment:
Are you intending override the getters/setters so that they fetch from rawSingletons/rawCollections instead? Otherwise this doesn't prevent inappropriate data access. Particles should not be able to modify or retrieve fields that are filtered out.

Reply:
Nvrmind, I see this is handled by the SCHEMA filtering in EntityBase.

Reply:
I'm not sure I understand how inappropriate data access would happen here. We only want to expose fields / properties specified by the TV constraint to developers. However, we want to copy data not specified by the constraint under-the-hood. 

In my understanding, this means modifying the (de)serialize methods, but not the getters/setters.

Reply:
If we point the getters/setters to the raw* maps, then particles *would* have access to "filtered-out" data.

Reply:
Ah, I got your "nvm" comment late :)